sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -1013,6 +1660,119 @@ public void onFailure(@NonNull Call<Void> call, @NonNull Throwable t) {
         }
     }
 
+    private BlobSetTagsResponse setBlobTagsWithRestResponseIntern(String containerName,
+                                                                  String blobName,
+                                                                  Integer timeout,
+                                                                  String versionId,
+                                                                  String iftags,
+                                                                  Map<String, String> tags,
+                                                                  String version,
+                                                                  String requestId,
+                                                                  CancellationToken cancellationToken,
+                                                                  CallbackWithHeader<Void, BlobSetTagsHeaders> callback) {
+
+        cancellationToken = cancellationToken == null ? CancellationToken.NONE : cancellationToken;
+
+        final String comp = "tags";
+
+        List<BlobTag> blobTagSet = null;
+        if (tags != null) {
+            blobTagSet = new ArrayList<>(tags.size());
+            for (Map.Entry<String, String> entry : tags.entrySet()) {
+                blobTagSet.add(new BlobTag().setKey(entry.getKey()).setValue(entry.getValue()));
+            }
+        }
+        BlobTags blobTags = new BlobTags();
+        blobTags.setBlobTagSet(blobTagSet);
+
+        RequestBody tagsBody;
+        try {
+            tagsBody = RequestBody.create(MediaType.get("application/xml; charset=utf-8"),
+                serializerAdapter.serialize(blobTags, SerializerFormat.XML));
+        } catch (IOException ioe) {
+            if (callback != null) {
+                callback.onFailure(ioe, null);
+
+                return null;
+            } else {
+                throw new RuntimeException(ioe);
+            }
+        }
+
+        Call<ResponseBody> call = service.setBlobTags(containerName,
+            blobName,
+            timeout,
+            versionId,
+            null, // TODO: calculate Md5?
```

Comment:
We probably need to consider this in [the MD5 PR](https://github.com/Azure/azure-sdk-for-android/pull/414).

Reply:
On second thought, this isn't actually possible for us at this point because it's the md5 of the payload, but we can't calculate that until after the body is serialized, at which point it's out of our control. We could consider adding an interceptor that does that, but it would be a separate PR I think.